cli/azd/pkg/commands/pipeline/pipeline.go

```diff
@@ -58,14 +64,20 @@ type ScmProvider interface {
 		remoteName string,
 		branchName string,
 		console input.Console) (bool, error)
+	//Hook function to allow SCM providers to handle scenarios after the git push is complete
+	postGitPush(ctx context.Context,
+		gitRepo *gitRepositoryDetails,
+		remoteName string,
+		branchName string,
+		console input.Console) error
 }
 
 // CiProvider defines the base behavior for a continuous integration provider.
 type CiProvider interface {
```

Comment:
I'm still not a fan of not exporting the functions in the interface to be consumable by outside packages.